spectrometer.py

```diff
@@ -1,303 +1,331 @@
+    csv.close()
+
+
+def export_diagram(name, normalized_results):
+    antialias = 4
+    w = 600 * antialias
+    h2 = 300 * antialias
+
+    h = h2 - 20 * antialias
+    sd = Image.new('RGB', (w, h2), (255, 255, 255))
+    draw = ImageDraw.Draw(sd)
+
+    w1 = 380.0
+    w2 = 780.0
+    f1 = 1.0 / w1
+    f2 = 1.0 / w2
+    for x in range(0, w, 1):
+        # Iterate across frequencies, not wavelengths
+        lambda2 = 1.0 / (f1 - (float(x) / float(w) * (f1 - f2)))
+        c = wavelength_to_color(lambda2)
+        draw.line((x, 0, x, h), fill=c)
+
+    pl = [(w, 0), (w, h)]
+    for wavelength in normalized_results:
+        wl = float(wavelength)
+        x = int((wl - w1) / (w2 - w1) * w)
+        # print wavelength,x
+        pl.append((int(x), int((1 - normalized_results[wavelength]) * h)))
+    pl.append((0, h))
+    pl.append((0, 0))
+    draw.polygon(pl, fill="#FFF")
+    draw.polygon(pl)
+
+    font = ImageFont.truetype('/usr/share/fonts/truetype/lato/Lato-Regular.ttf', 12 * antialias)
+    draw.line((0, h, w, h), fill="#000", width=antialias)
+
+    for wl in range(400, 1001, 10):
+        x = int((float(wl) - w1) / (w2 - w1) * w)
+        draw.line((x, h, x, h + 3 * antialias), fill="#000", width=antialias)
+
+    for wl in range(400, 1001, 50):
+        x = int((float(wl) - w1) / (w2 - w1) * w)
+        draw.line((x, h, x, h + 5 * antialias), fill="#000", width=antialias)
+        wls = str(wl)
+        tx = draw.textsize(wls, font=font)
+        draw.text((x - tx[0] / 2, h + 5 * antialias), wls, font=font, fill="#000")
+
+    # save chart
+    sd = sd.resize((int(w / antialias), int(h / antialias)), Image.ANTIALIAS)
+    output_filename = name + "_chart.png"
+    sd.save(output_filename, "PNG", quality=95, optimize=True, progressive=True)
+
+
+def main():
+    # 1. Take picture
+    camera = picamera.PiCamera()
+    name = sys.argv[1]
+    shutter = int(sys.argv[2])
+    raw_filename = take_picture(camera, name, shutter)
+    im = Image.open(raw_filename)
+
+    # 2. Get picture's aperture
+    pic_pixels = im.load()
+    aperture = find_aperture(pic_pixels, im.size[0], im.size[1])
+
+    # 3. Draw aperture and scan line
+    spectrum_angle = 0.03
+    draw = ImageDraw.Draw(im)
+    draw_aperture(aperture, draw)
+    draw_scan_line(aperture, draw, spectrum_angle)
+
+    # 4. Draw graph on picture
+    wavelength_factor = 0.892  # 1000/mm
+    # wavelength_factor=0.892*2.0*600/650 # 500/mm
+    results, max_result = draw_graph(draw, pic_pixels, aperture, spectrum_angle, wavelength_factor)
+
+    # 5. Inform user of issues with exposure
+    inform_user_of_exposure(max_result)
+
+    # 6. Save picture with overlay
+    save_image_with_overlay(im, name)
+
+    # 7. Normalize results for export
+    normalized_results = normalize_results(results, max_result)
+
+    # 8. Save csv of results
+    export_csv(name, normalized_results)
+
+    # 9. Generate spectrum diagram
+    export_diagram(name, normalized_results)
+
+
+main()
```

Comment:
Starting from the outside in...

I'd go ahead and put this in a `if __name__ == '__main__':` here.